drivers/disk/sdmmc_stm32.c

```diff
@@ -146,6 +190,68 @@ static int stm32_sdmmc_clock_disable(struct stm32_sdmmc_priv *priv)
 				 (clock_control_subsys_t *)&priv->pclken);
 }
 
+static int stm32_sdmmc_configure_dma(DMA_HandleTypeDef *handle, struct sdmmc_dma_stream *dma)
```

Comment:
Miss a 
```
#ifdef STM32_SDMMC_USE_DMA
```
?

Reply:
> Miss a
> 
> ```
> #ifdef STM32_SDMMC_USE_DMA
> ```
> 
> ?

My thinking is that since it never will be called(the function call is #ifdef'ed) the linker will remove that code anyway, it would be redundant. I'm not sure if the Zephyr project have a different policy/code standard regarding that?

Reply:
The problem is about the HAL DMA headers that might not be available on some series.
In Kconfig, we're actually using:
```
	select USE_STM32_HAL_DMA if (SOC_SERIES_STM32L4X || SOC_SERIES_STM32F7X || SOC_SERIES_STM32F4X)
```
Since this was done earlier than your PR this should be done due to a dependency inside the HAL itself.
So, on F4/L4/F7, no issues indeed.
On other series, we might face compilation issues, unless this is modified as well.
To me the safest is to protect this piece of code as well for now. This could be removed eventually once DMA support will be extended to other series as well.

Reply:
Makes sense. I will fix it!